.travis.yml

```diff
@@ -39,42 +40,60 @@ matrix:
     env: INSTALL_DEPENDS="pip==10.0.1 setuptools==30.3.0"
 
 before_install:
+  - python -m pip install --upgrade pip virtualenv
+  - virtualenv --python=python /tmp/venv
+  - source /tmp/venv/bin/activate
+  - python --version
+  - pip --version
   - python -m pip install --upgrade $INSTALL_DEPENDS
-  - pip install "flake8<3.0" flake8-putty
+  - |
+    if [ "$INSTALL_TYPE" == "pip" ]; then
+        PACKAGE="."
+    elif [ "$INSTALL_TYPE" == "sdist" ]; then
+        python setup.py sdist
+        PACKAGE="$( ls dist/*.tar.gz )"
+    elif [ "$INSTALL_TYPE" == "wheel" ]; then
+        python setup.py bdist_wheel
+        PACKAGE="$( ls dist/*.whl )"
+    fi
 
 install:
   - |
     if [ "$INSTALL_TYPE" == "install" ]; then
         python setup.py install
     elif [ "$INSTALL_TYPE" == "develop" ]; then
         python setup.py develop
-    elif [ "$INSTALL_TYPE" == "pip" ]; then
-        pip install .
-    elif [ "$INSTALL_TYPE" == "sdist" ]; then
-        python setup.py sdist
-        pip install dist/*.tar.gz
-    elif [ "$INSTALL_TYPE" == "wheel" ]; then
-        python setup.py bdist_wheel
-        pip install dist/*.whl
+    else
+        pip install $PACKAGE
     fi
   - |
-    if [ "$CHECK_TYPE" == "style" ]; then
-        travis_retry pip install "flake8<3.0" flake8-putty
+    INTENDED_VERSION="$(python -c 'import versioneer; print(versioneer.get_version())')"
+    pushd /tmp
+    INSTALLED_VERSION="$(python -c 'import niworkflows; print(niworkflows.__version__)')"
+    python -c 'import niworkflows; print(niworkflows.__file__)'
+    echo "Intended: $INTENDED_VERSION"
+    echo "Installed: $INSTALLED_VERSION"
+    test "$INTENDED_VERSION" == "$INSTALLED_VERSION"
+    popd
+  - if [ "$CHECK_TYPE" = "install" ]; then exit 0; fi
```

Comment:
Run the version check on all cases, and then short circuit if we're just checking the installation.